### PR TITLE
Fix `const_fn_trait_ref_impl`, add test for it

### PR DESCRIPTION
**Pull request by onestacked**
#99943 broke `#[feature(const_fn_trait_ref_impl)]`, this PR fixes this and adds a test for it.

r? @fee1-dead

### CODE REVIEW

**Comment by compiler-errors** [comment]
Whoops, thanks. 

@bors r+

**Comment by compiler-errors** [comment]
Actaully, one nit

@bors r-

File: src/test/ui/consts/fn_trait_refs.rs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// run-pass
```
Comment (compiler-errors) [inline src/test/ui/consts/fn_trait_refs.rs:1]: This does not need to be run-pass. `check-pass` will work just fine.

**Comment by chriss0612** [comment]
> Whoops, thanks.
I should probably have had a test in my original PR

File: src/test/ui/consts/fn_trait_refs.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+const fn main() {
+    const fn one() -> i32 {
+        1
+    };
+    const fn two() -> i32 {
+        2
+    };
+
+    // FIXME(const_cmp_tuple)
+    let test_one = test_fn(one);
```
Comment (chriss0612) [inline src/test/ui/consts/fn_trait_refs.rs:72]: Working on a PR for this, will fix this in the pr that is merged last

**Comment by chriss0612** [comment]
should probably be rollup=always

**Pull request merged**